### PR TITLE
Add NGen3 

### PR DESCRIPTION
**Pull request by Thishyaketh**
## This PR Fixes Issues in uploading NGen 3 to Transformers

### CODE REVIEW

**Comment by Thishyaketh** [comment]
@ArthurZucker and @Rocketknight1 
Hope it works this time!

File: src/transformers/models/ngen3/modular_ngen3.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        # Residual connection for attention
+        residual = x
+        x = self.ln1(x)
+        x = self.attn(x)
+        x = residual + x
+        # Residual connection for feedforward
+        residual = x
+        x = self.ln2(x)
+        x = self.mlp(x)
+        return residual + x
```
Comment (ArthurZucker) [inline src/transformers/models/ngen3/modular_ngen3.py:96]: can you try to align variable names / layer names with `LlamaDecoderLayer` here?

File: src/transformers/models/ngen3/modular_ngen3.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+class CausalSelfAttention(nn.Module):
```
Comment (ArthurZucker) [inline src/transformers/models/ngen3/modular_ngen3.py:98]: this should be completely equivalent to a variation of LlamaAttention with fused qkv. Nothing very new and the mask should not be saved there

**Comment by ArthurZucker** [comment]
@Thishyaketh we need a proper description of the PR with:
- original checkpiont release link
- original checkpoint paper implementing this
and something like this to introduce the model. 
Make sure you use `transformers-cli add-new-model-like` + try to explain what is different in your model compared to llama? 🤗

**Comment by Thishyaketh** [comment]
Thank you @ArthurZucker sir for the feedback I will make all the changes required and the paper + model check point link with difference between Llama and NGen 3
